include/tvm/runtime/logging.h

```diff
@@ -409,6 +416,68 @@ inline bool DebugLoggingEnabled() {
   return state == 1;
 }
 
+/*! \brief Helpers for \p VerboseLoggingEnabled. Exposed for unit testing only. */
+std::unordered_map<std::string, int> ParseTvmLogDebugSpec(const char* opt_spec);
+bool VerboseEnabledInMap(const std::string& filename, int level,
+                         const std::unordered_map<std::string, int>& map);
+
+/*!
+ * \brief Returns true if a VLOG statement in \p filename is enabled by the \p TVM_LOG_DEBUG
+ * environment variable for logging at verbosity \p level.
+ *
+ * Filenames are canonicalized to be w.r.t. the src/ dir of the TVM tree. (VLOG's should not
+ * appear under include/).
+ *
+ * To enable file \p relay/foo.cc up to level 2 and \p ir/bar.cc for level 0 only set:
+ * \code
+ * TVM_LOG_DEBUG="1;relay/foo.cc=2;ir/bar.cc=0;"
+ * \endcode
+ *
+ * To enable all files up to level 3 but disable \p ir/bar.cc set:
+ * \code
+ * TVM_LOG_DEBUG="1;*=2;ir/bar.cc=-1;"
```

Comment:
This usage looks like not user friendly, is there any other better way to do this? for example
TVM_LOG_DEBUG_LEVEL_2="relay/foo.cc;"
TVM_LOG_DEBUG_LEVEL_0="ir/bar.cc;"

Reply:
There are many possible designs. I settled on re-using the existing TWM_LOG_DEBUG since
 - it is one control surface
 - I often modules on and off during debugging and having one var to redefine makes that easy

Reply:
According the logic, TVM_LOG_DEBUG now have a hiding strict format assumptions, these setting  like "*=2;1;" or "*=2;1", "*=1;" will be illegal,   I don't see any necessary here to introduce such complexity for environment variable content.

Reply:
I definitely  agree with you the ergonomics are horrible! Using an env variable at all is already unfortunate but forced on us by the python-is-the-driver convention for TVM. But I think this is all very temporary and 'for developers only' while we figure out a more structured approach to controlling, capturing and redirecting logging, including both 'for debug' logging as well as logging which may be relevant to downstream customers running tvm in a production environment. 
(+ @jroesch since we were chatting about this today).

At this point the CI is green -- do you feel strongly enough we should not proceed as is?

Reply:
If we agree that is a problem , that sound not make sense not to fix it,  and better to avoid a temporary solution that in some case  will become a permanent solution, as a user for this VLOG feature, I like the idea filtering in file level, but  I think such  configuration method is unnecessary complicated and error-prone, if we can better to find a more easy to use method. 
about CI is green, I can understand the CI consume time and sometime it also random fail, to get it become green not very easy, but from my point view we should better to  more focus on code logic instead of CI.

Reply:
@huajsj i agree this is a bit more complex, but it is backwards-compatible. i also don't think it's especially complex given what it does. 

it seems like a more structured, frontend-agnostic interface might be a PackedFunc like:

```
TVM_REGISTER_GLOBAL("set_vlog_level").set_body_typed([](const std::string& file_name, int level) {
  my_global_map[file_name] = level
}
```

I'm totally open to adding this. However, I'd like to point out a couple of advantages of the environment variable route @mbs-octoml proposes here.
- use of a single variable or config string: it's very easy to determine exactly the VLOG configuration of TVM. you just `echo ${TVM_LOG_DEBUG}`. if multiple variables are used, the command becomes more complex: `export | grep TVM_LOG_DEBUG_LEVEL_`
- use of environment variable: We make heavy use of `multiprocessing` or `PopenPool`, and have many different frontends. i'd argue that VLOG level configuration is a TVM developer-facing interface. If we use a command-line flag or script to configure logging, then every time a program is launched with a different entry point, a parser would need to be written to convert some command-line flag into a set of `get_function("set_vlog_level")(file_name, level)` calls. Each script may have a different way to parse out the various vlog levels, so if a TVM C++ developer is trying to help a user debug a problem in `libtvm.so`, they first would need to determine how to configure TVM VLOG levels. `TVM_LOG_DEBUG` is a more foolproof route. Additionally, we would need to find a way to thread this down through `multiprocessing`/`PopenPool` so that this configuration is propagated to e.g. autotuning Builder/Runner.

I agree with your point that temporary solutions can tend to become permanent. In this case I'd argue that the complex part of this PR is in parsing the environment variable and not in the change to the logging system. Given the advantages listed above, I think I'd still want to keep this functionality even if we overhaul the fundamental way logging works here. I also don't think this mechanism has to be kept particularly backwards-compatible--we can continue to improve it as we consider bigger changes to logging. What do you think?

Reply:
thanks @areusch, @mbs-octoml for the follow up, that make sense.

Reply:
Along the same line as 'we have many frontends' I'm also trying to support writing C++ unit tests, and obviously being able to control logging for any test is vital. So any python-only control surface would not work for that case.

Reply:
As far as syntax is concerned:
 - obviously the legacy TVM_LOG_DEBUG=1 behavior must be kept. Tat will turn on DLOG but leave the implied default VLOG levels for all files as -1.
 - but for finer control the '1' start seems to be throwing folks off, so I'll drop that. Eg
TVM_LOG_DEBUG="foo/bar.cc=3" means  enable DLOG and VLOG for that file at levels 0  to 3.

Reply:
@mbs-octoml does that mean there's no way to do global + per-file at once?

Reply:
ok i see. i'm okay with this change if you also print a warning when someone tries to do `src/*.cc=3`